Can't have ':' in device path

I use a pass-through virtual machine with virtio input devices. The problem is that these devices have `:`s in their filenames which Kanata doesn't seem to handle.
`(defcfg linux-dev /dev/input/by-path/pci-0000:12:00.0-event-kbd)`
```
15:25:43 [INFO] kanata v1.0.7 starting
15:25:43 [INFO] NOTE: kanata was compiled to never allow cmd
15:25:43 [INFO] config parsed
15:25:43 [INFO] Created device "/dev/input/event11"
15:25:43 [INFO] Sleeping for 2s. Please release all keys and don't press additional ones.
15:25:45 [INFO] entering the processing loop
15:25:45 [INFO] entering the event loop
15:25:45 [ERROR] Failed to open the input keyboard device. Make sure you've added kanata to the `input` group. E: No such file or directory (os error 2)
15:25:45 [ERROR] main got error failed to open keyboard device: No such file or directory (os error 2)
```
I also tried using `\:` and `":"`, but got the exact same error.

I'm assuming this is because of #62, which separates devices with `:`.
Using a symlink to the device circumvents the issue and Kanata works perfectly fine.
`$ ln -s /dev/input/by-path/pci-0000\:12\:00.0-event-kbd ./input`
`(defcfg linux-dev ./input)`

--- a/cfg_samples/kanata.kbd
+++ b/cfg_samples/kanata.kbd
@@ -20,8 +20,13 @@
   ;; is preferable; I recall reading that it's less likely to change names on you,
   ;; but I didn't find any keyboard device in there in my VM.
   linux-dev /dev/input/by-path/platform-i8042-serio-0-event-kbd
+
   ;; If you want to read from multiple devices, separate them by `:`.
   ;; linux-dev /dev/input/<dev1>:/dev/input/<dev2>
+  ;;
+  ;; If you have a colon in your device path, add a backslash before it so that
+  ;; kanata does not parse it as multiple devices.
+  ;; linux-dev /dev/input/path-to\:device
 
   ;; Windows doesn't need any input/output configuration entries; however, there
   ;; must still be a defcfg entry.

--- a/src/oskbd/linux.rs
+++ b/src/oskbd/linux.rs
@@ -29,7 +29,11 @@ impl KbdIn {
         match KbdIn::new_linux(dev_paths) {
             Ok(s) => Ok(s),
             Err(e) => {
-                log::error!("Failed to open the input keyboard device. Make sure you've added kanata to the `input` group. E: {}", e);
+                log::error!(
+                    "Failed to open the input keyboard devices {:?}. Make sure the user running kanata is part of the input and uinput groups. Error: {}" ,
+                    parse_dev_paths(dev_paths),
+                    e,
+                );
                 Err(e)
             }
         }
@@ -38,12 +42,12 @@ impl KbdIn {
     fn new_linux(dev_paths: &str) -> Result<Self, std::io::Error> {
         let mut devices = HashMap::new();
         let poll = Poll::new()?;
-        for (i, dev_path) in dev_paths.split(':').enumerate() {
-            let mut kbd_in_dev = Device::open(dev_path)?;
 
-            // NOTE: This grab-ungrab-grab sequence magically
-            // fix an issue with a Lenovo Yoga trackpad not working.
-            // No idea why this works.
+        for (i, dev_path) in parse_dev_paths(dev_paths).iter().enumerate() {
+            let mut kbd_in_dev = Device::open(&dev_path)?;
+
+            // NOTE: This grab-ungrab-grab sequence magically fixes an issue with a Lenovo Yoga
+            // trackpad not working. No idea why this works.
             kbd_in_dev.grab()?;
             kbd_in_dev.ungrab()?;
             kbd_in_dev.grab()?;
@@ -318,6 +322,31 @@ impl Symlink {
             }
         });
     }
+}
+
+fn parse_dev_paths(paths: &str) -> Vec<String> {
+    let mut all_paths = vec![];
+    let mut full_dev_path = String::new();
+    let mut dev_path_iter = paths.split(':').peekable();
+    while let Some(dev_path) = dev_path_iter.next() {
+        if dev_path.ends_with('\\') && dev_path_iter.peek().is_some() {
+            full_dev_path.push_str(dev_path.trim_end_matches('\\'));
+            full_dev_path.push(':');
+            continue;
+        } else {
+            full_dev_path.push_str(dev_path);
+        }
+        all_paths.push(full_dev_path.clone());
+        full_dev_path.clear();
+    }
+    all_paths
+}
+
+#[test]
+fn test_parse_dev_paths() {
+    assert_eq!(parse_dev_paths("h:w"), ["h", "w"]);
+    assert_eq!(parse_dev_paths("h\\:w"), ["h:w"]);
+    assert_eq!(parse_dev_paths("h\\:w\\"), ["h:w\\"]);
 }
 
 impl Drop for Symlink {

